pkg/replicator/replicator.go

```diff
@@ -54,7 +55,16 @@ func NewReplicator(user apiv1alpha1.SystemUser, pass, host string, port int32) (
 	return &dbImpl{db}, nil
 }
 
+func (d *dbImpl) StopReplication() error {
+	_, err := d.db.Exec("STOP REPLICA")
+	return err
+}
+
 func (d *dbImpl) ChangeReplicationSource(host, replicaPass string, port int32) error {
+	if err := d.StopReplication(); err != nil {
```

Comment:
I would do this explicitly in `reconcileUsers` and `StartReplication` functions

Reply:
It was unnecessary, deleted.